File: backend/src/medicore/presentation/schemas/organization.py
"""Organization schemas."""

from __future__ import annotations

from pydantic import BaseModel


class UpdateOrganizationRequest(BaseModel):
    legal_name: str | None = None
    tax_id: str | None = None
    timezone: str | None = None
    plan: str | None = None
    seat_limit: int | None = None


class AddLocationRequest(BaseModel):
    name: str
    address: str | None = None
    is_primary: bool = False


class UpdateLocationRequest(BaseModel):
    name: str | None = None
    address: str | None = None


class LocationResponse(BaseModel):
    id: str
    name: str
    address: str | None
    is_primary: bool


class OrganizationResponse(BaseModel):
    id: str
    legal_name: str
    tax_id: str
    slug: str
    timezone: str
    plan: str
    seat_limit: int
    locations: list[LocationResponse]

File: frontend/src/api/organization.ts
import { api } from './client'
import type { Organization } from '@/types'

export const organizationApi = {
  get: () => api.get<Organization>('/organization').then((r) => r.data),

  update: (payload: Partial<Pick<Organization, 'legal_name' | 'tax_id' | 'timezone' | 'plan'>>) =>
    api.patch<Organization>('/organization', payload).then((r) => r.data),

  addLocation: (payload: { name: string; address?: string; is_primary?: boolean }) =>
    api.post<Organization>('/organization/locations', payload).then((r) => r.data),

  updateLocation: (id: string, payload: { name?: string; address?: string }) =>
    api.patch<Organization>(`/organization/locations/${id}`, payload).then((r) => r.data),
}
